src/librustc_mir/build/scope.rs

```diff
@@ -665,29 +659,35 @@ fn build_diverge_scope<'tcx>(tcx: &TyCtxt<'tcx>,
                           build_free(tcx, unit_temp, free_data, target));
             free_data.cached_block = Some(into);
             into
-        }
-    };
+        };
+    }
 
-    if let Some((previous_block, previous_span, previous_value)) = previous {
-        // Finally, branch into that just-built `target` from the `previous_block`.
-        cfg.terminate(previous_block,
-                      scope.id,
-                      previous_span,
-                      TerminatorKind::Drop {
-                          value: previous_value,
-                          target: target,
-                          unwind: None
-                      });
-        last_drop_block.unwrap()
-    } else {
-        // If `previous.is_none()`, there were no drops in this scope – we return the
-        // target.
-        target
+    // Next, build up the drops. Here we iterate the vector in
```

Comment:
Weren’t we supposed to execute drops first and then frees second (drops may depend on memory still being allocated, but frees do not otherwise care about things being dropped)?